lib/src/realm_class.dart

```diff
@@ -496,8 +496,13 @@ class Realm implements Finalizable {
   }
 
   static set logger(Logger value) {
+    if (_logger == value) {
+      return;
+    }
+
     _logger.clearListeners();
-    _logger = (value is _RealmLogger) ? value : _RealmLogger(value);
+    _logger = value;
+    _logger.onLevelChanged.listen((logLevel) => realmCore.loggerSetLogLevel(logLevel ?? RealmLogLevel.off, scheduler.nativePort));
```

Comment:
We should track this subscription and cancel it when a new logger is set, so that we don't have another logger which log level can change the one in core?

Ie. to avoid 
```dart
final old = Realm.logger;
Realm.logger = CustomLogger();
old.logLevel = Level.ALL;
```

Reply:
good catch. fixing